### PR TITLE
params: Remove separate CarParams from each daemon

### PR DESCRIPTION
**Pull request by nuwandavek**
Many parameter estimation daemons like `paramsd`, `torqued` , `magd` etc need to reject cached params if the device has switched cars. This logic was handled separately (and differently) in each daemon. Adding this param to `controlsd` will enable each such daemon to have a generic "check if this is a different car".

- New param called `CarParamsPrevRoute` for all daemons to use
- Generic sigint handler for all `ParameterEstimator`s, with the base class definition

### CODE REVIEW

**Comment by nuwandavek** [comment]
> Can we do the rest of the changes in this PR?

Done.

File: selfdrive/controls/controlsd.py [hunk 1]
```diff
@@ -121,6 +121,11 @@ def __init__(self, CI=None):
       safety_config.safetyModel = car.CarParams.SafetyModel.noOutput
       self.CP.safetyConfigs = [safety_config]
 
+    # Write previous route's CarParams
+    prev_cp = self.params.get("CarParamsPersistent")
+    if prev_cp is not None:
```
Comment (adeebshihadeh) [inline selfdrive/controls/controlsd.py:126]: don't you also want to check if it's the same car?

Comment (nuwandavek) [reply selfdrive/controls/controlsd.py:126]: The checks are handled in the daemons (there may be different checks). I just need access to the previous carParam from one source.

**Comment by nuwandavek** [comment]
@adeebshihadeh it looks like all three are necessary:
-  `{"CarParams", CLEAR_ON_MANAGER_START | CLEAR_ON_ONROAD_TRANSITION}` : used pretty much in all the daemons
- `{"CarParamsCache", CLEAR_ON_MANAGER_START}`: is necessary for fingerprinting
- `{"CarParamsPersistent", PERSISTENT}`: Used for all the UI toggle etc.

All are overwritten to the current `CP` during `controlsd` init.

**Review by adeebshihadeh — APPROVED** [review]

**Pull request merged**